core/variant/variant_parser.cpp

```diff
@@ -1051,11 +1051,11 @@ Error VariantParser::parse_value(Token &token, Variant &value, Stream *p_stream,
 			bool got_bracket_token = false;
 			if (builtin_types.has(token.value)) {
 				array.set_typed(builtin_types.get(token.value), StringName(), Variant());
-			} else if (token.value == "Resource" || token.value == "SubResource" || token.value == "ExtResource") {
+			} else if (token.value == Variant("Resource") || token.value == Variant("SubResource") || token.value == Variant("ExtResource")) {
```

Comment:
> Thanks for opening a PR. Most of these changes don't make sense from the coding perspective though. Those types that you are making explicit should be comparable already, why would it be incompatible with C++ 2020?

over.match.oper 12.4.2.3

TL;DR
Rewritten candidates include the set of comparison operators with their parameters reversed. Instead of simply implicitly converting one type to another, which probably isn't what you want, in C++20 there are two candidates for these operators in the overload set and neither is a better match than the other.